source/extensions/filters/listener/proxy_protocol/proxy_protocol_header.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+#include "common/common/assert.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace ListenerFilters {
+namespace ProxyProtocol {
+
+// See https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt for definitions
+
+constexpr char PROXY_PROTO_V1_SIGNATURE[] = "PROXY ";
+constexpr uint32_t PROXY_PROTO_V1_SIGNATURE_LEN = 6;
+constexpr char PROXY_PROTO_V2_SIGNATURE[] = "\x0d\x0a\x0d\x0a\x00\x0d\x0a\x51\x55\x49\x54\x0a";
+constexpr uint32_t PROXY_PROTO_V2_SIGNATURE_LEN = 12;
+constexpr uint32_t PROXY_PROTO_V2_HEADER_LEN = 16;
+constexpr uint32_t PROXY_PROTO_V2_VERSION = 0x2;
+constexpr uint32_t PROXY_PROTO_V2_ONBEHALF_OF = 0x1;
+constexpr uint32_t PROXY_PROTO_V2_LOCAL = 0x0;
+
+constexpr uint32_t PROXY_PROTO_V2_AF_INET = 0x1;
+constexpr uint32_t PROXY_PROTO_V2_AF_INET6 = 0x2;
+constexpr uint32_t PROXY_PROTO_V2_AF_UNIX = 0x3;
+
+struct WireHeader {
+  WireHeader(size_t el, Network::Address::IpVersion pv, Network::Address::InstanceConstSharedPtr ra,
```

Comment:
nit: please expand "pv", "ra", "la", etc. I understand it's assigned immediately below, but it's much harder to read quickly. I would make the struct members postfixed with `_` for this reason.